run_unittests.py

```diff
@@ -3315,6 +3324,36 @@ def test_apple_bitcode_modules(self):
         self.build()
         self.run_tests()
 
+    @skipIfNoPkgconfig
+    def test_uninstalled_usage_external_library(self):
+        '''
+        Test that uninstalled usage of an external library (from the system or
+        PkgConfigDependency) works. On Linux/BSD/macOS it tests if RPATHs are
+        set correctly.
+
+        TODO: On Windows, this can test whether PATH is set properly
+
+        The system library is found with cc.find_library() and pkg-config deps.
+        '''
+        oldprefix = self.prefix
+        # Install external library so we can find it
+        testdir = os.path.join(self.unit_test_dir, '33 external, internal library rpath', 'external library')
+        installdir = self.installdir
+        self.prefix = installdir
+        self.init(testdir)
+        self.build()
+        self.install(use_destdir=False)
+        self.prefix = oldprefix
+        # New builddir for the consumer
+        self.new_builddir()
+        os.environ['LIBRARY_PATH'] = os.path.join(installdir, self.libdir)
```

Comment:
Should this be `LD_LIBRARY_PATH`?

Reply:
`LIBRARY_PATH` adds that path to the compiler's path; so `gcc --print-search-dirs` will show it under `libraries` on Linux/BSD, and on macOS it'll be silently added to the list of paths searched after the system.

The reason why we do this is so that it can be found with `cc.find_library()` without using `dirs:`.